WordPress/Classes/ViewRelated/Post/PageMenuViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+
+final class PageMenuViewModel: AbstractPostMenuViewModel {
+
+    private let page: Page
+    private let isJetpackFeaturesEnabled: Bool
+    private let isBlazeFlagEnabled: Bool
+
+    var buttonSections: [AbstractPostButtonSection] {
+        [
+            createPrimarySection(),
+            createSecondarySection(),
+            createBlazeSection(),
+            createTrashSection()
+        ]
+    }
+
+    init(
+        page: Page,
+        isJetpackFeaturesEnabled: Bool = JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled(),
+        isBlazeFlagEnabled: Bool = BlazeHelper.isBlazeFlagEnabled()
+    ) {
+        self.page = page
+        self.isJetpackFeaturesEnabled = isJetpackFeaturesEnabled
+        self.isBlazeFlagEnabled = isBlazeFlagEnabled
+    }
+
+    private func createPrimarySection() -> AbstractPostButtonSection {
+        var buttons = [AbstractPostButton]()
+
+        if !page.isFailed {
+            buttons.append(.view)
```

Comment:
(Production issue) https://github.com/wordpress-mobile/WordPress-iOS/issues/14428

Reply:
Hmmm! So https://github.com/wordpress-mobile/WordPress-iOS/issues/6037 (preview trashed posts) was opened in 2016, while https://github.com/wordpress-mobile/WordPress-iOS/issues/14428 (hide view action for trashed posts) was opened in 2020. 

On web, the view action is hidden for trashed posts and pages. I propose doing the same for apps.

<img width="1153" alt="Screenshot 2023-11-01 at 12 59 42" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/6323a61a-76b9-4d0f-bdeb-b2263d08b0e2"> | <img width="1214" alt="Screenshot 2023-11-01 at 12 59 51" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/4751eb61-96f4-4eea-8e44-8669a523e71c">
-- | --

Reply:
I'm not sure it needs to be a menu item. I would suggest making the cells tappable. If you tap on a trashed post – generate a preview, same as for drafts.